lib/sinon/behavior.js

```diff
@@ -377,11 +377,11 @@ function createAsyncVersion(syncFnName) {
 }
 
 // create asynchronous versions of callsArg* and yields* methods
-for (var method in proto) {
+Object.keys(proto).forEach(function (method) {
     // need to avoid creating anotherasync versions of the newly added async methods
     if (proto.hasOwnProperty(method) && method.match(/^(callsArg|yields)/) && !method.match(/Async/)) {
```

Comment:
I've pushed a review commit that removes it. Once we're happy with the PR, I'll rebase and squash this commit